bigquery/inserter.go

```diff
@@ -19,16 +19,28 @@ import (
 	"errors"
 	"fmt"
 	"reflect"
+	"strings"
 
 	"cloud.google.com/go/internal/trace"
 	bq "google.golang.org/api/bigquery/v2"
+	"google.golang.org/api/googleapi"
 )
 
 // NoDedupeID indicates a streaming insert row wants to opt out of best-effort
 // deduplication.
 // It is EXPERIMENTAL and subject to change or removal without notice.
 const NoDedupeID = "NoDedupeID"
 
+// ErrRequestMalformed is surfaced for cases where the response from the backend
+// indicates a generic error, but is unstructured and ambiguous.
+//
+// For streaming insert requests, this often indicates that either the row
+// data is excessively large, or that the batch factor is too high (too many rows
+// sent as part of a single insert).  However, the error response provides no
+// further diagnostic information, and may indicate problems other than size such
+// as QPS or fundamental problems with the payload.
+var ErrRequestMalformed = fmt.Errorf("your client has issued a malformed or illegal request")
```

Comment:
prefer `errors.New` if you aren't using wrapping

Reply:
Do you expect people to type switch on this error and handle this case specially? Just trying to understand if this needs to be exported.

Reply:
I'm coming around to the viewpoint of not adding a custom error, and simply augmenting the package documentation to be more explicit.  It provides little value and doesn't reduce ambiguity sufficiently.